source/Calamari.AzureWebApp.NetCoreShim/WebDeploymentExecutor.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Calamari.AzureWebApp.NetCoreShim.Retry;
+using Microsoft.Web.Deployment;
+using Newtonsoft.Json;
+using Serilog;
+
+namespace Calamari.AzureWebApp.NetCoreShim
+{
+    public class WebDeploymentExecutor
+    {
+        static readonly LimitedExponentialRetryInterval RetryIntervalForAzureOperations = new LimitedExponentialRetryInterval(5000, 30000, 2);
+        readonly ILogger logger;
+
+        public WebDeploymentExecutor(ILogger logger)
+        {
+            this.logger = logger;
+        }
+
+        public async Task Execute(Program.SyncOptions options)
+        {
+            var retry = new RetryTracker(4,
+                                         timeLimit: TimeSpan.MaxValue,
+                                         retryInterval: RetryIntervalForAzureOperations);
+            while (retry.Try())
+            {
+                try
+                {
+                    var changeSummary = DeploymentManager
+                                        .CreateObject("contentPath", options.SourceContentPath)
+                                        .SyncTo(
+                                                "contentPath",
+                                                options.DestinationContentPath,
+                                                DeploymentOptions(options),
```

Comment:
premature optimization - but can the DeploymentOptions and DeploymentsyncOptions be calculated once - given they only depend on the program's options?